File: ADA3/DataSource.java
//https://docs.oracle.com/javase/specs/jls/se20/jls20.pdf
package java;

import java.io.File;
import java.io.IOException;
import java.util.Scanner;

public class DataSource {
    private File dataFile;
    private Bank bank;

    public DataSource(File dataFile) {
        this.dataFile = dataFile;
        this.bank = new Bank();
    }

    public void loadData() throws IOException {
        if (!dataFile.exists()) {

        } else {
            System.out.println("Existe");
            Scanner input = new Scanner(dataFile);
            int numAccounts;
            char accountType;

            int numCustomers = input.nextInt();

            for (int i = 0; i < numCustomers; i++) {
                String firstName = input.next();
                String lastName = input.next();
                bank.addCustomer(firstName, lastName, i);
                Customer customer = bank.getCustomer(i);
                numAccounts = input.nextInt();
                System.out.println(numCustomers);

                while (numAccounts-- > 0) {
                    accountType = input.next().charAt(0);
                    float initBalance = Float.parseFloat(input.next());

                    switch (accountType) {
                        case 'S':
                            float interestRate = Float.parseFloat(input.next());
                            customer.addAccount(new SavingAccount(initBalance, interestRate));
                            break;
                        case 'C':
                            float overdraftProtection = Float.parseFloat(input.next());
                            customer.addAccount(new CheckingAccount(initBalance, overdraftProtection));
                            break;
                        default:
                            break;
                    }
                }

            }
        }
    }

    public void generateCustomerReport() {
        System.out.println("\t\t\tCUSTOMERS REPORT");
        System.out.println("\t\t\t================");

        for (int i = 0; i < bank.getNumCustomers(); i++) {
            Customer customer = bank.getCustomer(i);
            System.out.println("Customer:" + customer.getLastName() + ", " + customer.getFirstName());
        }

    }
}

File: ADA3/DataSource.vb
'https://learn.microsoft.com/es-es/dotnet/visual-basic/programming-guide/language-features/declared-elements/declared-element-names#rules
'https://learn.microsoft.com/es-es/dotnet/visual-basic/programming-guide/program-structure/naming-conventions
'https://learn.microsoft.com/es-es/dotnet/visual-basic/programming-guide/concepts/object-oriented-programming#classes-and-objects
'https://learn.microsoft.com/en-us/dotnet/api/system.io.fileinfo?view=net-7.0
'https://www.aprenderaprogramar.es/index.php?option=com_content&view=article&id=257:select-case-visual-basic-switch-segun-caso-hacer-ejemplos-y-ejercicios-resueltos-con-to-is-cu00325a

'https://learn.microsoft.com/es-mx/dotnet/fundamentals/code-analysis/style-rules/naming-rules
'https://learn.microsoft.com/es-es/dotnet/visual-basic/language-reference/statements/dim-statement

'los métodos públicos usan pascal case
'ttodos loc campos constantes no públicos son s_camelCase. 

    Public Sub LoadData()
        If Not dataFile.Exists Then

        Else
            Console.WriteLine("Existe")
            Dim input As New StreamReader(dataFile.FullName)
            Dim numAccounts As Integer
            Dim accountType As Char

            Dim numCustomers As Integer = Convert.ToInt32(input.ReadLine())

            For i As Integer = 0 To numCustomers - 1
                Dim firstName As String = input.ReadLine()
                Dim lastName As String = input.ReadLine()
                bank.AddCustomer(firstName, lastName, i)
                Dim customer As Customer = bank.GetCustomer(i)
                numAccounts = Convert.ToInt32(input.ReadLine())
                Console.WriteLine(numCustomers)

                While numAccounts > 0
                    accountType = Convert.ToChar(input.ReadLine())
                    Dim initBalance As Single = Convert.ToSingle(input.ReadLine())

                    Select Case accountType
                        Case "S"c
                            Dim interestRate As Single = Convert.ToSingle(input.ReadLine())
                            customer.AddAccount(New SavingAccount(initBalance, interestRate))
                            Exit Select
                        Case "C"c
                            Dim overdraftProtection As Single = Convert.ToSingle(input.ReadLine())
                            customer.AddAccount(New CheckingAccount(initBalance, overdraftProtection))
                            Exit Select
                        Case Else
                            Exit Select
                    End Select
                    numAccounts -= 1
                End While
            Next

        End If
    End Sub